src/lib.rs

```diff
@@ -1,820 +1,651 @@
-//! Raw binding for RocksDB.
-//!
-//! This is just a thin binding to the RocksDB C API, mostly generated
-//! by rust-bindgen and lightly edited. This is intended to underpin a
```

Comment:
My concern is larger than that.

Thus far, I've been incrementally updating this file to reflect changes in the RocksDB `c.h` API, so that changes to this files are more or less 1:1 with changes in `c.h`.

Your set of changes represent a total rewrite. I'm not sure:
- which changes are the result of a different generator
- which changes are bugfixes
- which changes are aesthetic/formatting fixes
- which changes are platform porting fixes
- which changes are RocksDB API changes

It's good that you've started to break it down into separate commits, but it's still far from "one concern per commit".

And as a result, I'm not sure how to maintain this in future. Can I just regenerate it mechanically? If I did, what things would I need to port into the newly generated version to maintain compatibility?

You're also changing the API presented by this crate, which is something that other crates depend on, so you're going to cause downstream breakage. If you're going to do that, there needs to be a good/convincing reason for it.

I would prefer to see:
- the minimal set of changes needed to update to the new RocksDB
- the minimal set of changes needed to get it working on Mac OS
- the minimal set of changes needed to fix the test bugs
- anything else

as separate commits which we can review individually.